index.js

```diff
@@ -152,7 +152,13 @@ module.exports = class Corestore extends EventEmitter {
 
     while (this.cores.has(id)) {
       const existing = this.cores.get(id)
-      if (existing.opened && !existing.closing) return { from: existing, keyPair, auth }
+      if (existing.opened && !existing.closing) {
+        if (opts.encryptionKey && !existing.encryptionKey) {
+          await existing.close()
```

Comment:
need to grok if this has very bad side-effects, ie would this close open sessions?

Reply:
Probably.. the alternative is considering this https://github.com/hypercore-protocol/hypercore-next/issues/95 then.